src/Common/initial_condition.jl

```diff
@@ -281,3 +281,100 @@ function cloudy_initial_condition(pdists, ip, k = 1)
 
     return merge(ip, (; moments = moments, pdists = pdists, cloudy_moments_zero))
 end
+
+function p3_initial_condition(::Type{FT}, thermo_params, q_init, N_init, z; F_rim, F_liq, z_top, ice_start) where {FT}
+
+    _ρ_r::FT = FT(900) # fixed initial ρ_r
+
+    # if ice_start, start with small ice bump
+    # otherwise, introduce particles solely through
+    # the boundary in advection_tendency!()
+    _z_band = z_top * FT(0.2)
+    has_ice(z) = ifelse(ice_start, z > (z_top - _z_band), false)
+    ice_bump(χ, z) = χ * cos(z - (z_top - 0.5 * _z_band))
+
+    # P3 state variables (minus B_rim -- computed below)
+    q_ice::FT = has_ice(z) ? ice_bump(q_init, z) : FT(0)
+    N_ice_kg::FT = has_ice(z) ? ice_bump(N_init, z) : FT(0)
+    q_liqonice::FT = F_liq * q_ice
+    q_rim::FT = F_rim * (q_ice - q_liqonice)
+
+    # 2M warm rain state variables
+    q_liq::FT = FT(0)
+    N_liq::FT = FT(0)
+    q_rai::FT = FT(0)
+    N_rai::FT = FT(0)
+
+    # q_tot - single p3 category + 2M categories
+    q_tot::FT = q_ice + q_liq + q_rai
```

Comment:
Maybe we could add some water vapor to this setup. Such that q_tot = q_vap + q_liq + q_ice?

Reply:
Sure! Would you say I should add some initial q_vap using the same cosine initial condition as the ice_bump (line 294) uses? If so, is there a reasonable amplitude I should use?
I'm guessing water vapor will be used in ice nucleation but that otherwise it will hang out where it starts (as long as we don't prescribe an updraft velocity)

Reply:
Ah—update: I could also use the same profile of water vapor which I just noticed that's used in the normal 1D testing setup. I'm guessing that would be the best way forward.